### ISSUE TITLE
GitHub の一般的なエンジニアとしての使われ方事例紹介

### ISSUE THREAD

**Issue by Gucchiy** [icat-lab/seminar_design_hakuba#13]
GitHub 自体は g4n のような使われ方をしているケースは今のところほぼないので、まずは、OSS の実態として、どのようにこれらツールが運用されているのか、on going で開発が進められているページを紹介してはどうでしょう？ Issues や Project 、できれば Code の branch tree が実際に使われている様子が見えると良いかな、と

**Comment by yofukami** [comment]
@Gucchiy はい。いいと思います。やるとすると初回でしょうか？これ、実際に @Gucchiy がコミットされているプロジェクトのリポジトリを紹介いただくなんて形でいかがでしょう？

**Comment by Gucchiy** [comment]
@yofukami はい、了解です。いえ、2、3、できればみなさんが使っているもののプロジェクトが紹介できるとベストなので探しておきます。前期思ったのですが、あまり GitHub がもたらしたムーブメントを実感していただけてないな・・と思ったので、そのあたりの"熱量"が分かっていただければよいかと思います。ので、ちょっとそのあたりが見えればいいかな、と。今や Civc-tech 側に戻った（？）Cocoa もなかなかいい例になってますね。
https://github.com/cocoa-mhlw/cocoa/issues
https://github.com/cocoa-mhlw/cocoa/projects/5

**Comment by yofukami** [comment]
@Gucchiy 春学期は、試しに輪読で始めてみた、というものですので、確かにGitHubによるムーブメントは紹介できてなかったですよね、、、。これは大きな反省点です。今学期は、グループワーク、つまりコラボレーションがメインのゼミですので、今回はど真ん中だと思っています。ちなみに、COCOAもいいですけど、  @Gucchiy が関わっていらっしゃる　*地域活性化*
 関連のプロジェクトで適切なものはありませんか？今回は地域がテーマになりますので。

**Comment by Gucchiy** [comment]
@yofukami 以前関わっていた、地域密着型の社会復帰支援 NPO さんの事例もご紹介はしてみましょう。ただ、エンジニアは二人、NPO 側3名のコラボレーションだったので、ブランチなどは活用されず、Project と Issue がメインなので、「ムーブメント」に主眼があるとするとちょっとイマイチです。どこかのでっかいプロジェクトといっしょにご紹介するのが良いかな、と思います。ちなみに、NPTech (Civic Tech の NPO 版）も、なんだか、各支援の取りまとめと、コミュニティ・ビルディングが大変になっていて、ぜんぜん、GitHub 活用まで至ってない感じですね

**Comment by yofukami** [comment]
@Gucchiy 確かに、コラボレーションがムーブメントになるとなると、、、なかなか難しいですよね。Code forのブリゲイドでも、そこまで多人数が活発なのは、、、

**Comment by yofukami** [comment]
と思って、つらつらとろくに投稿したことがないCode for JapanのSlackをつらつら視ていたら、こういうのがあるんですね、、

[codeforjapan/remote\-patient\-monitoring\-api](https://github.com/codeforjapan/remote-patient-monitoring-api)

**Comment by Gucchiy** [comment]
Code For Japan のプロジェクトをご紹介する場合、ちょっとツライのが、多くのプロジェクトで、ツールの Project が活用されてないんですよね。去年リリースされたばかりなので・・
でも、探せば何か良いのあるかもですね。大勢でなくても良いんですが、Project / Issue / Branch が活用されている方がいいでしょうね。後、分かりやすいヤツ

**Comment by yofukami** [comment]
確かに、、、。
他のも色々見ていますが、projectですよね、、、

**Comment by Gucchiy** [comment]
実は MS 系のはむっちゃ活用されているんです。そりゃそうかもですが。
例えば、もはや、MS が作っているとは誰も気が付かないほど普及しつつある TypeScript のプロジェクトは以下。JavaScript に型を厳密にした言語で、Google も使うようになりました。最近の Node.js コードは TypeScript のものが多いですね。しかし、ちょっとエンジニア寄り過ぎの案件なので、これじゃダメだと思いますが：
https://github.com/microsoft/TypeScript/projects/13

ちなみに、MS は元々、[Azure + Visual Studio の製品として DevOps を売っていて](https://azure.microsoft.com/ja-jp/services/devops/)、Issue みたいなウェブベースの管理ツールも SaaS で提供していたのですが、あっさり諦めて、社内でさえツールは GitHub に移行しつつあります。ウェブベースで、結構よくできてました。Git リポジトリも作れます。

**Comment by yofukami** [comment]
@Gucchiy 元々DXをテーマに掲げたシラバスでしたので、TypeScriptのプロジェクトも、背景を説明すればかなり受けそうな気もします。このような最新のコラボレーション手法を、あらゆる場面で活用するための手法開発だという打ち出しはしておいたほうがいいかもしれません。

**Comment by Gucchiy** [comment]
うーん、TypeScript 登場の純粋な背景って、オブジェクト指向とプロトタイプ指向という超マニアックな言語仕様の線なんですよ。技術的ニーズというよりは、コード作成上のニーズ、まあ、結果的に設計品質に繋がるんですが、これには「アーキテクチャ」の良し悪しという視点がないと理解できないようなー。まあ、検討してみますが

**Comment by Gucchiy** [comment]
本件、一旦、9/15(WED) に使った以下の解説で完了で良いでしょうかね。こちらも売り物シートがいっぱい入っちゃってるので、配布等はご遠慮くださいませ。
[20210914_Scrum_github.pdf](https://github.com/icat-lab/seminar_design_hakuba/files/7233817/20210914_Scrum_github.pdf)
